shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/presentation/detail/PlantDetailComponent.kt

```diff
@@ -1,33 +1,46 @@
 package com.vml.tutorial.plantshop.plants.presentation.detail
 
 import com.arkivanov.decompose.ComponentContext
+import com.vml.tutorial.plantshop.core.utils.componentCoroutineScope
+import com.vml.tutorial.plantshop.plants.data.PlantsRepository
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
 class PlantDetailComponent(
     plant: Plant,
     componentContext: ComponentContext,
-    private val onNavigateToBack: ()-> Unit
-): ComponentContext by componentContext {
+    private val plantsRepository: PlantsRepository,
+    private val onNavigateToBack: () -> Unit
+) : ComponentContext by componentContext {
     private val _state = MutableStateFlow(PlantDetailState(plant))
     val state: StateFlow<PlantDetailState> = _state.asStateFlow()
 
+    init {
+        _state.update { it.copy(isFavourite = it.plant.isFavorite) }
+    }
+
     fun onEvent(event: PlantDetailEvent) {
-        when(event) {
+        when (event) {
             PlantDetailEvent.CheckoutPlant -> TODO()
             PlantDetailEvent.OnFavouriteClick -> {
-                _state.update { it.copy(isFavourite = !it.isFavourite)}
+                _state.update { it.copy(isFavourite = !it.plant.isFavorite) }
```

Comment:
Ahh resolved my previous comment here... :D so if you don't mind you can remove this line and add something similar into this Component ->

    val state: StateFlow<PlantDetailState> = _state.combine(plantsRepository.getFavorites()) { state, favourites ->
        state.copy(
            isFavourite = favourites.any { it.id == state.plant.id }
        )
    }.stateIn(componentCoroutineScope(), SharingStarted.Lazily, _state.value)
    
   This will update uiState whenever user toggledFavouriteStatus

Reply:
Refactored :)